ui/redux/actions/settings.js

```diff
@@ -117,7 +118,25 @@ export function doSaveCustomWalletServers(servers) {
 }
 
 export function doSetClientSetting(key, value) {
-  return dispatch => {
+  return (dispatch, getState) => {
+    const {
+      settings: { daemonSettings },
+    } = getState();
+    const { share_usage_data: shareSetting } = daemonSettings;
+    const isSharingData = shareSetting || IS_WEB;
+
+    if (key === SETTINGS.SHOW_REPOSTS && isSharingData) {
```

Comment:
I think this should just live in the component onClick handler

Right now it looks like I would need to opt in to sharing data if I want to add the tag to remove those emails.